dev-requirements.txt

```diff
@@ -357,25 +361,23 @@ ptyprocess==0.7.0
     # via
     #   pexpect
     #   terminado
-py==1.11.0
-    # via pytest
-pycodestyle==2.9.1
+pycodestyle==2.10.0
     # via nexusutils (pyproject.toml)
 pycparser==2.21
     # via cffi
 pygments==2.13.0
     # via
     #   ipython
     #   nbconvert
-pylint==2.15.3
+pylint==2.17.2
```

Comment:
why do we have to change all these dependencies?

Reply:
while including 'nanonispy' and rerun pip-compile it has created the file itself satisfying  the dependencies condition.